### PR TITLE
Refactoring for Swift 3.1.

### PR DESCRIPTION
**Pull request by andersio**

### CODE REVIEW

**Comment by mdiep** [comment]
That test doesn't want to pass on this branch. 😕

File: ReactiveCocoaTests/InterceptingSpec.swift [hunk 1]
```diff
@@ -8,6 +8,8 @@ import CoreGraphics
 
 class InterceptingSpec: QuickSpec {
 	override func spec() {
+		ForwardInvocationTestObject.swizzleForTesting()
```
Comment (NachoSoto) [inline ReactiveCocoaTests/InterceptingSpec.swift]: I'd put this in `beforeAll`. If these tests aren't gonna run there's no point in calling this function.

File: ReactiveCocoaTests/UIKit/UIButtonSpec.swift [hunk 2]
```diff
@@ -7,6 +7,8 @@ import enum Result.NoError
 
 class UIButtonSpec: QuickSpec {
 	override func spec() {
+		UIControl.swizzleForTesting()
```
Comment (NachoSoto) [inline ReactiveCocoaTests/UIKit/UIButtonSpec.swift]: Ditto

File: ReactiveCocoaTests/InterceptingSpec.swift (hunk 1 above)
Comment (andersio) [inline ReactiveCocoaTests/InterceptingSpec.swift]: `beforeAll` is not available in Quick IIRC.

Comment (NachoSoto) [inline ReactiveCocoaTests/InterceptingSpec.swift]: Oh I think its `beforeSpec` now.

File: ReactiveCocoaTests/InterceptingSpec.swift [hunk 3]
```diff
@@ -824,13 +828,23 @@ class InterceptingSpec: QuickSpec {
 	}
 }
 
+extension DispatchQueue {
```
Comment (NachoSoto) [inline ReactiveCocoaTests/InterceptingSpec.swift]: This can be private

**Comment by andersio** [comment]
#3446 seems to have a better approach. Closing for now.

**Pull request closed without merging**